resources/aws_ec2.go

```diff
@@ -1014,6 +994,65 @@ func (obj *AwsEc2Res) prependName() string {
 	return AwsPrefix + obj.GetName()
 }
 
+// waitForStopAndTerminate waits for the instance to stop and waits for it to
+// terminate. If either waiter returns, the instance state is checked, and an
+// awsEc2Event is returned.
+func (obj *AwsEc2Res) waitForStopAndTerminate(ctx context.Context, waitInput *ec2.DescribeInstancesInput) (awsEc2Event, error) {
+	wg := sync.WaitGroup{}
+	wgCtx, wgCancel := context.WithCancel(ctx)
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		defer wgCancel()
+		if err := obj.client.WaitUntilInstanceStoppedWithContext(wgCtx, waitInput); err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				if aerr.Code() == request.CanceledErrorCode || aerr.Code() == request.WaiterResourceNotReadyErrorCode {
+					return
+				}
+			}
+			select {
+			case obj.awsChan <- &chanStruct{
+				err: errwrap.Wrapf(err, "unknown error waiting for instance to stop"),
+			}:
+			case <-obj.closeChan:
+			}
+		}
+	}()
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		defer wgCancel()
+		if err := obj.client.WaitUntilInstanceTerminatedWithContext(wgCtx, waitInput); err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				if aerr.Code() == request.CanceledErrorCode || aerr.Code() == request.WaiterResourceNotReadyErrorCode {
+					return
+				}
+			}
+			select {
```

Comment:
This shouldn't be part of this function if you agree with the above comment about matching the AWS API.